src/sentry/notifications/notification_action/types.py

```diff
@@ -136,17 +136,21 @@ def create_rule_instance_from_action(
         if not features.has(
             "organizations:workflow-engine-ui-links", detector.project.organization
         ):
-            if job.workflow_id is None:
-                raise ValueError("Workflow ID is required when triggering an action")
+            # If test event, just set the legacy rule id to -1
+            if job.workflow_id == -1:
+                data["actions"][0]["legacy_rule_id"] = -1
+            else:
+                if job.workflow_id is None:
+                    raise ValueError("Workflow ID is required when triggering an action")
```

Comment:
nit: I don't think this needed to change locations. Having this check at the beginning makes sense to me.